collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/AudioRecordingTest.java

```diff
@@ -0,0 +1,175 @@
+package org.odk.collect.android.feature.formentry;
+
+import android.Manifest;
+import android.app.Application;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import androidx.test.espresso.intent.rule.IntentsTestRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.GrantPermissionRule;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.MainMenuActivity;
+import org.odk.collect.android.support.TestDependencies;
+import org.odk.collect.android.support.TestRuleChain;
+import org.odk.collect.android.support.pages.FormEndPage;
+import org.odk.collect.android.support.pages.GeneralSettingsPage;
+import org.odk.collect.android.support.pages.MainMenuPage;
+import org.odk.collect.android.support.pages.SaveOrIgnoreDialog;
+import org.odk.collect.audiorecorder.recording.AudioRecorderViewModel;
+import org.odk.collect.audiorecorder.recording.AudioRecorderViewModelFactory;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.odk.collect.android.support.FileUtils.copyFileFromAssets;
+
+@RunWith(AndroidJUnit4.class)
+public class AudioRecordingTest {
```

Comment:
I'd been trying to resist writing an Espresso test due to the feature being isolated within the widget but making recording work inline requires collaboration between a few different things (the widget and the activity for instance) so it started to feel too disjointed just writing tests at the component level. 

I do like where these tests have ended up however as they capture the high level navigation features without having to deal with any of the detail of audio recording or playback.